### PR TITLE
gifs only!

### PR DESCRIPTION
**Pull request by ryanjones**
- Append 'gif' to search query - more likely to return gifs
- Only ever return gifs (discard anything that could possibly NOT be a gif (.png.gif for ex)
- Add tests/simplecov checking to make sure we have everything covered.

### CODE REVIEW

**Comment by mvandenbeuken** [comment]
:book:

File: lib/lita/handlers/animate.rb [hunk 1]
```diff
@@ -44,6 +52,27 @@ def safe_value
         safe = "active" unless ["active", "moderate", "off"].include?(safe)
         safe
       end
+
+      def valid_extension?(result)
+        if /^.*\.(jpg|JPG|png|PNG|jpeg|JPEG)/.match(result)
```
Comment (mvandenbeuken) [inline lib/lita/handlers/animate.rb]: what do you think about ```/^.*\.(jpg|png|jpeg)/i```?

Comment (ryanjones) [inline lib/lita/handlers/animate.rb]: I copy pasted this from SO. I'm a bad person and didn't think about what I was doing.

Comment (mvandenbeuken) [inline lib/lita/handlers/animate.rb]: wait a sec. what about ```/.*\.gif$/.match(result)```?

Comment (ryanjones) [inline lib/lita/handlers/animate.rb]: I want to return false if any of these extensions are found.

Comment (ryanjones) [inline lib/lita/handlers/animate.rb]: Sometimes rando junk like .png.gif comes in and it's not a real .gif

Comment (mvandenbeuken) [inline lib/lita/handlers/animate.rb]: ahh

Comment (ryanjones) [inline lib/lita/handlers/animate.rb]: if image (or even a possible image)
false
if gif
true
otherwise
nothing

Comment (mvandenbeuken) [inline lib/lita/handlers/animate.rb]: here we go I think! ```^((?!jpg|jpeg|png).)*\.gif$```

File: lib/lita/handlers/animate.rb [hunk 2]
```diff
@@ -44,6 +52,27 @@ def safe_value
         safe = "active" unless ["active", "moderate", "off"].include?(safe)
         safe
       end
+
+      def valid_extension?(result)
+        if /^.*\.(jpg|png|jpeg)/i.match(result)
```
Comment (mvandenbeuken) [inline lib/lita/handlers/animate.rb]: This should work as a one liner: /^((?!jpg|jpeg|png).)*\.gif$/i

File: lib/lita/handlers/animate.rb [hunk 3]
```diff
@@ -12,27 +14,33 @@ def self.default_config(handler_config)
       end
 
       def fetch(response)
```
Comment (mvandenbeuken) [inline lib/lita/handlers/animate.rb]: how do you feel about trying to break apart this method? It's a bit of a beast!

you could probably parcel off some methods into single-responsibility methods like an image randomizer, and maybe a response handler?

File: spec/lita/handlers/animate_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+describe Lita::Handlers::Animate, lita_handler: true do
```
Comment (mvandenbeuken) [inline spec/lita/handlers/animate_spec.rb]: :clap: :fireworks:  great tests!

**Comment by MathieuGilbert** [comment]
:hankey: on Mike's behalf

File: lib/lita/handlers/animate.rb (hunk 2 above)
Comment (ryanjones) [inline lib/lita/handlers/animate.rb]: :+1:

**Comment by ryanjones** [comment]
:wave:

File: lita-animate.gemspec [hunk 5]
```diff
@@ -1,8 +1,8 @@
 Gem::Specification.new do |spec|
   spec.name          = "lita-animate"
   spec.version       = "1.1.0"
-  spec.authors       = ["Michael van den Beuken", "Ruben Estevez", "Jordan Babe", "Mathieu Gilbert", "Ryan Jones", "Darko Dosenovic"]
-  spec.email         = ["michael.beuken@gmail.com", "ruben.a.estevez@gmail.com", "jorbabe@gmail.com", "mathieu.gilbert@ama.ab.ca", "ryan.michael.jones@gmail.com", "darko.dosenovic@ama.ab.ca"]
+  spec.authors       = ["Igor Adrov"]
```
Comment (MathieuGilbert) [inline lita-animate.gemspec]: But where's our credit?

**Comment by MathieuGilbert** [comment]
:elephant: :elephant: :elephant:

**Pull request merged**